src/modifiers/mod_remove_all_fields_from_classes.py

```diff
@@ -0,0 +1,51 @@
+from src import code_dom
+from src import utils
+
+
+# This modifier removes all the fields from the classes specified
+def apply(dom_root, class_names, add_dummy_field):
+    for class_element in dom_root.list_all_children_of_type(code_dom.DOMClassStructUnion):
+        if class_element.get_fully_qualified_name() not in class_names:
+            continue
+
+        fields = class_element.list_directly_contained_children_of_type(
+            code_dom.DOMFieldDeclaration)
+
+        # We want to also remove fields inside direct ifdefs
+        preprocessor_ifs = class_element.list_directly_contained_children_of_type(code_dom.DOMPreprocessorIf)
```

Comment:
This code shouldn't be necessary, because `list_directory_contained_children_of_type` is supposed to explicitly include children that are inside preprocessor conditionals.
...
I say "supposed" because that wasn't working (TIL that `__` name prefix in Python isn't just a polite hint that something is private, but the interpreter actively won't call overrides on functions with it!). Sorry!

I'll push a quick patch for this to mainline when I've finished looking through this PR, and then it should be safe to remove this bit.